src/framework/audio/internal/platform/win/wasapiaudioclient.cpp

```diff
@@ -560,11 +560,34 @@ void WasapiAudioClient::getSamples(uint32_t framesAvailable)
     uint8_t* data;
 
     uint32_t actualFramesToRead = framesAvailable;
-    uint32_t actualBytesToRead = actualFramesToRead * m_mixFormat->nBlockAlign;
+
+    // WASAPI: "nBlockAlign must be equal to the product of nChannels and wBitsPerSample divided by 8 (bits per byte)"
+    const uint32_t clientFrameSize = m_mixFormat->nBlockAlign;
+
+    // MuseScore assumes only 2 audio channels (same calculation as above to determine frame size)
+    const uint32_t muFrameSize = 2 * m_mixFormat->wBitsPerSample / 8;
 
     check_hresult(m_audioRenderClient->GetBuffer(actualFramesToRead, &data));
-    if (actualBytesToRead > 0) {
-        m_sampleRequestCallback(nullptr, data, actualBytesToRead);
+    if (actualFramesToRead > 0) {
+        // Based on the previous calculations, the only way that clientFrameSize will be larger than muFrameSize is
+        // if the client specifies more than 2 channels. MuseScore doesn't support this (yet), so we use a workaround
+        // where the missing channels are padded with zeroes...
+        if (clientFrameSize > muFrameSize) {
+            const size_t surroundBufferDesiredSize = actualFramesToRead * muFrameSize;
+            if (m_surroundAudioBuffer.size() < surroundBufferDesiredSize) {
+                m_surroundAudioBuffer.resize(surroundBufferDesiredSize);

```

Comment:
m_surroundAudioBuffer.resize(surroundBufferDesiredSize, 0); -- please add this for safety